mantidproject/mantid
# Add system test for opening the algorithm dialogs

**Description of work.**
This PR adds a system test that checks that all the different algorithm dialogs open ok with different workspaces in the ADS.

As mentioned in the issue, this test could take quite a long time if we are not careful. I have tried to reduce its run time in two ways:
    - The test will only open the most recent version of an algorithm (so 1000 algorithm dialogs are opened instead of 1300)
    - The tests for each workspace is in a different module to allow for the tests to be run concurrently

It would be good to hear from the reviewer if they think this second point is a good/bad idea. Not sure if its an ideal solution, so any other suggestions are welcome

Some algorithms have been excluded in the `self._exclude_algorithm` variables because they cause errors or crashes when opened. These algorithm dialogs will be fixed, and the variable removed, in a separate PR. See issue #34114

**To test:**
Check the test passes on each of the workspace type. (Re-run the test until you see that it has run on each of the workspace tests.

Fixes #34032 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
